### PR TITLE
Resolve some ng-dirty/ng-pristine issues.

### PR DESCRIPTION
**Pull request by marbletravis**
Resolves issue #41.

Resolved issues with ng-dirty/ng-pristine on focus/click by adding the ability to add silent event options allowing users to determine which events alter the dirty/pristine state. 

By default, I have it leaving it pristine on focus/click, not sure if we want to introduce that to everyone but it makes sense for me.

There are still some issues with it marking the input as dirty when you change the mask (see the demo). I looked at this, but couldn't see it right off the bat. I don't think it is the same issue as outlined in #41

Tests have been updated, let me know if you want it in its own section, but seemed to fit in the one I put it in.

Lastly, do I need to update/release, I assume that is done by a mod.

### CODE REVIEW

File: dist/mask.js [hunk 1]
```diff
@@ -424,7 +425,13 @@ angular.module('ui.mask', [])
                                 iElement.val(valMasked);
                                 
                                 scope.$apply(function() {
-                                    controller.$setViewValue(valUnmasked); // $setViewValue should be run in angular context, otherwise the changes will be invisible to angular and user code.
+                                    // $setViewValue/$viewValue should be run in angular context, otherwise the changes will be invisible to angular and user code.
+                                    if (linkOptions.silentEvents.indexOf(eventType) >= 0) {
```
Comment (NickTomlin) [inline dist/mask.js]: I think simplifying this `linkOptions.silentEvents.indexOf(eventType) !== -1` feels a little more idiomatic.

Also, Array.indexOf is ie9+ (I'm assuming we don't actually support below 9 so this is probably fine). Any thoughts @PowerKiKi?

**Comment by NickTomlin** [comment]
:jeans: 
This is great, thanks so much for taking this work on. I don't have release powers but @PowerKiKi can.

**Comment by PowerKiKi** [comment]
@marbletravis it seems you are trying to merge your master and that a few unrelated commits  were pushed. Could you please clean up the situation by removing non-related commits (and maybe squashing all related commits in a single commit) ? then i'll merge and release

**Comment by marbletravis** [comment]
@PowerKiKi I have squashed the commits.

Thanks,
Travis

**Comment by Tim91084** [comment]
@marbletravis @PowerKiKi  Could you take a look at my pull request and see if it accomplishes the same thing?  It may be a better alternative to the fix you are proposing.

**Comment by PowerKiKi** [comment]
@Tim91084 would you elaborate on what makes you think #57 would be better ?

**Comment by Tim91084** [comment]
On second thought, I guess I see the merit in suppressing $setViewValue always as a result of a click / focus.  Only issue I can think of is dragging and dropping some input into the text box, but that is likely not a common use case.  The only other issue I have is with the implementation.  We don't need the check for silent events within $scope.apply, do we?  We can avoid triggering a digest here.

When I was looking at the issue, I went back in the history to see who had removed "valAltered" and why they did it, and made a better attempt at fixing that issue specifically.  This actually wasn't an issue that long ago - it has been recently introduced.

**Comment by marbletravis** [comment]
Closing pull request, #57 addressed this better. However one part (the on blur) wasn't addressed by #57 (to my knowledge) and is now addressed in #70

**Pull request closed without merging**